src/librustc_mir/interpret/operand.rs

```diff
@@ -538,11 +538,10 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             Scalar::Ptr(ptr) => Scalar::Ptr(self.tag_static_base_pointer(ptr)),
             Scalar::Raw { data, size } => Scalar::Raw { data, size },
         };
+        let value = self.subst_and_normalize_erasing_regions_in_frame(val.val)?;
```

Comment:
After talking with @eddyb I cam to the conclusion that this is actually wrong. There is no reason to assume here that `val` came from the current frame.

Instead, whoever gets a constant out of the MIR of this frame should do this -- other callers won't need this.

Reply:
Yupp, only `eval_operand` should be doing this.
Come to think of it, I don't even think you need to allow calling `subst_and_normalize_erasing_regions_in_frame` without a frame, once this and `resolve` is fixed.

Reply:
> Yupp, only eval_operand should be doing this.

Right -- and https://github.com/rust-lang/rust/commit/721df8f558af5c583365b0c90e9b06df7cc25026 shows that we indeed got this wrong so far. I wonder what happens if we fix just that, if that's enough duck tape to make it still mostly work. ;)

Reply:
`eval_const_to_op` should probably get a comment similar to what I proposed for `resolve`, saying it is the callers responsibility to subst the argument into the interpreter context.